canonical/maas-ui
# feat(machines): update clone form machine search API

## Done

- update clone form to perform machine search with new API
- map system_id to id filter

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to machine listing
- Select a machine from the list
- Take action -> clone from
- Make sure you can search for a source machine and select it from the list
- Verify that clone errors are shown correctly with appropriate machine counts
- Verify that the attempt to perform action cloning sends a request with a correct filter selected and system_id for the source machine
- Once a clone action has failed for a selection of machines and their system_ids are returned from the back-end in error messages verify you can click the "Show" link that will display the machine listing filtered by the machines that have failed
- Repeat the steps above with various machine selections, including entire groups

## Fixes

Fixes: https://github.com/canonical/app-tribe/issues/1111

## Launchpad issue

Related Launchpad maas issue in the form `lp#number`.

## Backports

In general, please propose fixes against _main_ rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in main.

## Screenshots

It could be helpful to provide some screenshots to aid in QAing the change.
